src/common/src/com/microsoftopentechnologies/tooling/msservices/helpers/azure/AzureManagerImpl.java

```diff
@@ -0,0 +1,2160 @@
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+import com.microsoft.windowsazure.management.ManagementClient;
+import com.microsoft.windowsazure.management.compute.ComputeManagementClient;
+import com.microsoft.windowsazure.management.network.NetworkManagementClient;
+import com.microsoft.windowsazure.management.storage.StorageManagementClient;
+import com.microsoftopentechnologies.tooling.msservices.components.AppSettingsNames;
+import com.microsoftopentechnologies.tooling.msservices.components.DefaultLoader;
+import com.microsoftopentechnologies.tooling.msservices.components.PluginSettings;
+import com.microsoftopentechnologies.tooling.msservices.helpers.*;
+import com.microsoftopentechnologies.tooling.msservices.helpers.auth.AADManager;
+import com.microsoftopentechnologies.tooling.msservices.helpers.auth.AADManagerImpl;
+import com.microsoftopentechnologies.tooling.msservices.helpers.auth.UserInfo;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.rest.AzureAADHelper;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.rest.AzureCertificateHelper;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.rest.MobileServiceRestManager;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.rest.RestServiceManager.ContentType;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.rest.RestServiceManagerBaseImpl;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.rest.model.*;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.sdk.AzureSDKHelper;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.sdk.SDKRequestCallback;
+import com.microsoftopentechnologies.tooling.msservices.model.Subscription;
+import com.microsoftopentechnologies.tooling.msservices.model.ms.*;
+import com.microsoftopentechnologies.tooling.msservices.model.storage.StorageAccount;
+import com.microsoftopentechnologies.tooling.msservices.model.vm.*;
+import org.apache.commons.lang.NotImplementedException;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+import sun.misc.BASE64Decoder;
+import sun.misc.BASE64Encoder;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import java.io.*;
+import java.lang.reflect.Type;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class AzureManagerImpl implements AzureManager {
+    private static class ReadWriteLockBox {
```

Comment:
There were already other getUserLock and getSubscriptionLock methods that returned NotNull, and I didn't want to have two methods with the same name, with @NotNull and @Nullable.
I'll apply the suggested changes, changing the name to something more descriptive, like getUserLockIfPresent and getSubscriptionLockIfPresent.